test/addons-abi/test_buffer/test_buffer.cc

```diff
@@ -0,0 +1,141 @@
+  char *bufferData;
+  NAPI_CALL(env, napi_get_buffer_data(env, theBuffer, &bufferData));
+  bool stringsEqual = !strcmp(bufferData, theText);
+  napi_value returnValue;
+  NAPI_CALL(env, napi_create_boolean(env, stringsEqual, &returnValue));
+  NAPI_CALL(env, napi_set_return_value(env, info, returnValue));
+}
+
+NAPI_METHOD(staticBuffer) {
+  napi_value theBuffer;
+  NAPI_CALL(env, napi_create_buffer(env, (char *)theText,
+    sizeof(theText), noopDeleter, &theBuffer));
+  NAPI_CALL(env, napi_set_return_value(env, info, theBuffer));
+}
+
+NAPI_METHOD(bufferLength) {
+  int argc;
+  NAPI_CALL(env, napi_get_cb_args_length(env, info, &argc));
+  JS_ASSERT(env, argc == 1, "Wrong number of arguments");
+  napi_value theBuffer;
+  NAPI_CALL(env, napi_get_cb_args(env, info, &theBuffer, 1));
+  size_t result;
+  NAPI_CALL(env, napi_get_buffer_length(env, theBuffer, &result));
+  napi_value returnValue;
+  NAPI_CALL(env, napi_create_number(env, result, &returnValue));
+  NAPI_CALL(env, napi_set_return_value(env, info, returnValue));
+}
+
+void Init(napi_env env, napi_value exports, napi_value module) {
+  napi_propertyname propName;
+  napi_value theValue;
+
+  NAPI_CALL(env, napi_property_name(env, "theText", &propName));
+  NAPI_CALL(env, napi_create_string_utf8(env, theText, sizeof(theText), &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "newBuffer", &propName));
+  NAPI_CALL(env, napi_create_function(env, newBuffer, nullptr, &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "getDeleterCallCount", &propName));
+  NAPI_CALL(env, napi_create_function(env, getDeleterCallCount, nullptr,
+    &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "copyBuffer", &propName));
+  NAPI_CALL(env, napi_create_function(env, copyBuffer, nullptr, &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "bufferHasInstance", &propName));
+  NAPI_CALL(env, napi_create_function(env, bufferHasInstance, nullptr,
+    &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "bufferData", &propName));
+  NAPI_CALL(env, napi_create_function(env, bufferData, nullptr, &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "staticBuffer", &propName));
+  NAPI_CALL(env, napi_create_function(env, staticBuffer, nullptr, &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
+
+  NAPI_CALL(env, napi_property_name(env, "bufferLength", &propName));
+  NAPI_CALL(env, napi_create_function(env, bufferLength, nullptr, &theValue));
+  NAPI_CALL(env, napi_set_property(env, exports, propName, theValue));
```

Comment:
All these properties could be set up more cleanly with a single call to `napi_define_properties()`.